pkg/deploy/elbv2/load_balancer_manager.go

```diff
@@ -186,20 +193,41 @@ func (m *defaultLoadBalancerManager) updateSDKLoadBalancerWithSecurityGroups(ctx
 	}
 	desiredSecurityGroups := sets.NewString(awssdk.StringValueSlice(securityGroups)...)
 	currentSecurityGroups := sets.NewString(awssdk.StringValueSlice(sdkLB.LoadBalancer.SecurityGroups)...)
-	if desiredSecurityGroups.Equal(currentSecurityGroups) {
+
+	isEnforceSGInboundRulesOnPrivateLinkUpdated, currentEnforceSecurityGroupInboundRulesOnPrivateLinkTraffic, desiredEnforceSecurityGroupInboundRulesOnPrivateLinkTraffic := isEnforceSGInboundRulesOnPrivateLinkUpdated(resLB, sdkLB)
+	if desiredSecurityGroups.Equal(currentSecurityGroups) && !isEnforceSGInboundRulesOnPrivateLinkUpdated {
 		return nil
 	}
 
+	if !desiredSecurityGroups.Equal(currentSecurityGroups) {
```

Comment:
1. Nit, since there is only one change call technically, i'd prefer to use a single log call
var changeDescriptions []string
if !desiredSecurityGroups.Equal(currentSecurityGroups) {
   changeDescriptions = append(changeDescriptions, "changeSecurityGroups", changeSecurityGroupsDesc)
}
req := &elbv2sdk.SetSecurityGroupsInput{
		LoadBalancerArn: sdkLB.LoadBalancer.LoadBalancerArn,
		SecurityGroups:  securityGroups,
}

if ! isEnforceSGInboundRulesOnPrivateLinkUpdated {
   changeDescriptions = append(changeDescriptions, "changeEnforceSecurityGroupInboundRulesOnPrivateLinkTraffic", changeEnforceSecurityGroupInboundRulesOnPrivateLinkTrafficDesc)
   req.EnforceSecurityGroupInboundRulesOnPrivateLinkTraffic: awssdk.String(desiredEnforceSecurityGroupInboundRulesOnPrivateLinkTraffic),
}
m.logger.Info("modifying loadBalancer security group settings",
			"stackID", resLB.Stack().StackID(),
			"resourceID", resLB.ID(),
			"arn", awssdk.StringValue(sdkLB.LoadBalancer.LoadBalancerArn),
			...changeDescriptions)
2. per the current log, `req.EnforceSecurityGroupInboundRulesOnPrivateLinkTraffic` is only set when there is a change. So if the security group changed but `isEnforceSGInboundRulesOnPrivateLinkUpdated` is false, then req.EnforceSecurityGroupInboundRulesOnPrivateLinkTraffic will be nil, what's the API behavior under such case, will it untouch the existing EnforceSecurityGroupInboundRulesOnPrivateLinkTraffic settings on LB or reset it to default value(something we don't want).

Reply:
1. Yes, we can switch to use a single log call.

Reply:
2. Good question, it will untouch the existing EnforceSecurityGroupInboundRulesOnPrivateLinkTraffic settings